packages/devtools_app/test/ansi_up_test.dart

```diff
@@ -145,8 +145,8 @@ void main() {
 
       final detailsComputer = computedDetails == null
           ? null
-          : () =>
-              Future.delayed(const Duration(seconds: 1), () => computedDetails);
+          : () => Future.delayed(
+              const Duration(seconds: 1), () => computedDetails!);
```

Comment:
hoist computedDetails into local

Reply:
computedDetails is local

Reply:
@bkonyi interesting, so does the flow analysis not work with ternary operators? after the `:` we know `computedDetails` is not null

Reply:
I'm not sure this is an issue with the analysis not working for ternary operators, but is probably due to the asynchrony involved or simply the variable capture by the closure creation.

For example, we can invalidate the null check in this simple example:

```dart
void main() {
  String? foo = 'bar';
  print(foo + 'baz'); // No null check needed since we instantiated foo.
  Future.delayed(const Duration(seconds: 1, () => print(foo)));
  foo = null;
}
```

The output from this is:

```
barbaz
null
```

Reply:
Confirmed this isn't a ternary operator issue since the following example works fine:

```dart
String? foo() => 'foo';

void main() {
  final x = foo();
  // Try to append to 'x' to double check it's not null.
  print(x == null ? null : x + 'bar');
}
```

Which outputs: `foobar`